src/main/java/seedu/duke/command/SearchModuleCommand.java

```diff
@@ -0,0 +1,164 @@
+package seedu.duke.command;
+
+import seedu.duke.utils.State;
+import seedu.duke.utils.Storage;
+import seedu.duke.utils.Ui;
+import seedu.duke.model.Module;
+import seedu.duke.parser.Parser;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+public class SearchModuleCommand extends Command {
+    public static final String FORMAT = "search /code [MODULE_CODE] /title [MODULE_TITLE] "
+            + "/level [LEVEL] /sem [SEMESTER]";
+    public static final String COMMAND_WORD = "search";
+
+    // private String toSearchModuleCode;
+    private Map<String, String> params;
+    private String toSearchModuleCode;
+    private String toSearchModuleTitle;
+    private String toSearchLevel;
+    private String toSearchSemester;
+
+    private Logger logger;
+
+    public static final String SUBSYSTEM_NAME = "SearchModuleCommand";
+
+    public SearchModuleCommand(String input) {
+        super(input.split("\\s"));
+        params = Parser.parseParams(input);
+    }
+
+    @Override
+    public void execute(State state, Ui ui, Storage storage) {
+        assert state != null : "List of lessons should not be null";
+        logger = Logger.getLogger(SUBSYSTEM_NAME);
+        logger.log(Level.INFO, "Loading search module command");
+
+        toSearchModuleCode = params.getOrDefault("code", null);
+        toSearchModuleTitle = params.getOrDefault("title", null);
+        toSearchLevel = params.getOrDefault("level", null);
+        toSearchSemester = params.getOrDefault("sem", null);
+
+        List<Module> searchResult = filterModuleSearch(toSearchModuleCode, toSearchLevel,
+                toSearchSemester, toSearchModuleTitle);
+
+        ui.addMessage("Search Result:");
+
+        if (searchResult.size() == 0) {
+            ui.addMessage("No module found");
+        } else {
+            ui.addMessage("Total " + searchResult.size() + " module(s) found\n");
+            for (int i = 0; i < searchResult.size(); i++) {
+                ui.addMessage(searchResult.get(i).moduleCode + " " + searchResult.get(i).title);
+            }
+            ui.addMessage("\nTo get full details of the module, type 'get <module code>'");
+        }
+
+        ui.displayUi();
+    }
+
+    /**
+     * Filter the module list based on input module level.
+     *
+     * @param module the module object
+     * @param level  the level that user input
+     * @return true if module level matches input level
+     */
+    static boolean isSameModuleLevel(Module module, String level) {
+        // get the first int embedded in the module code
+        String moduleCode = module.moduleCode;
+        int moduleLevel = (Integer.parseInt(moduleCode.replaceAll("[^0-9]", ""))) / 1000;
```

Comment:
can put the level calculation in model/Module.java

Reply:
shifted